packages/api-sdk/src/medical/models/fhir.ts

```diff
@@ -86,30 +86,25 @@ export const resourceSchema = z.array(z.enum(resourceTypeForConsolidation));
 export const consolidationConversionType = ["html", "pdf", "json"] as const;
 export type ConsolidationConversionType = (typeof consolidationConversionType)[number];
 
-export const consolidatedFilterSchema = z.object({
-  resources: z.string().optional(),
-  dateFrom: z.date().optional(),
-  dateTo: z.date().optional(),
+export const getConsolidatedFiltersSchema = z.object({
+  resources: z.enum(resourceTypeForConsolidation).array().optional(),
+  dateFrom: z.string().optional(),
+  dateTo: z.string().optional(),
```

Comment:
Can't these be `Date`? Then, if we `.parse()` the response in the SDK, the users would get that typed to date as well, instead of string.

If not, this would introduce a breaking change on the SDK, right?

If would be good to have code that uses the SDK function so we know whether it breaks the SDK or not in compile time.

Reply:
We could but the params we send from postman are a string and the response we send back to the cxs for the consolidated count is also a string

Reply:
Let's separate the issues:
1. the params we send from postman are a string
   - yeah, everything goes as string, we just need to parse them in the router
3. the response we send back to the cxs for the consolidated count is also a string
   - again, the API response is a string representation of the date, in ISO 8601
   - but another thing is how the SDK returns it form the call to the JS/TS function - we convert that `any` to our type, so it could be a parsed date to simplify the user's life
   - the format we represent the information inside the app should not be determined by the endpoint response
   - we'd just need to, on the router, convert the dates from `Date` to `string` in the ISO 8601 format
      - that's why we had the DTOs in the past, but we slowly moved away from it 🤷

Reply:
Gotcha i can update this but this isnt going to be a simple change. AFAIK we've been treating these as strings in various places

Reply:
Gotcha i can update this but its not going to be a simple change as this would change alot of files. We have all the consolidated endpoints set up to work with strings as well as the functions that are called within them. We had `consolidatedFilterSchema` that had the dates but AFAIK this was actually never being used

Reply:
I could just create a Resp type that has the types as Dates to return the user and internal we continue managing them as strings

Reply:
Yeah, it makes sense to keep as is. I see we're storing that on the DB as well, so defs a much larger thing.

WRT to updating the SDK, maybe we can ticket it as an improvement?

Reply:
https://github.com/metriport/metriport-internal/issues/1876